rpcs3/Emu/Io/RB3MidiKeyboard.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "Emu/Io/usb_device.h"
+
+#include <rtmidi_c.h>
+
+class usb_device_rb3_midi_keyboard : public usb_device_emulated
+{
+private:
+	u32 response_pos = 0;
+	bool buttons_enabled = false;
+	RtMidiInPtr midi_in;
+
+	// button states
+	// TODO: emulate velocity
+	struct
+	{
+		u8 count = 0;
+
+		bool cross = false;
+		bool circle = false;
+		bool square = false;
+		bool triangle = false;
+
+		bool start = false;
+		bool select = false;
+		bool overdrive = false;
+
+		bool dpad_up = false;
+		bool dpad_down = false;
+		bool dpad_left = false;
+		bool dpad_right = false;
+
+		std::array<bool, 25> keys{};
+		std::array<u8, 25> velocities{};
+
+		s16 pitch_wheel = 0;
+	} button_state;
+
+	void parse_midi_message(u8 msg[32], usz size);
+	void write_state(u8 buf[27]);
```

Comment:
```suggestion
	void write_state(u8* buf);
```

Reply:
I'll make this change but the rationale behind this was to make it clear that the function takes a 27 byte buffer.